src/solver/pad_constant/pad_constant_fwd_contiguous.cpp

```diff
@@ -0,0 +1,133 @@
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    if(!problem.IsSameType())
+    {
+        return false;
+    }
+
+    return true;
+}
+
+ConvSolution PadConstantFwdContiguous::GetSolution(
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+    auto ydims  = problem.GetYDesc().GetLengths();
+
+    auto input_dtype  = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetYDesc().GetType());
+
+    // for xgridsize: 5d -> 1d
+    size_t output_size = problem.GetYDesc().GetElementSize();
+
+    size_t xlocalsize = 1024;
+    // AlignUp blows up, because output_size can be > int_max. Lovely.
+    // size_t xgridsize  = AlignUp(output_size, xlocalsize);
+    size_t xgridsize =
+        (((static_cast<size_t>(output_size) + xlocalsize - 1) / xlocalsize) * xlocalsize);
+    size_t ylocalsize = 1;
+    size_t ygridsize  = 1;
+
+    auto kernel = KernelInfo{};
+
+    kernel.kernel_file = "MIOpenPadConstantFwd.cpp";
+    kernel.kernel_name = "PadConstantFwdContiguous";
+
+    // TODO: Actually understand how to use this properly
+    const auto build_params = KernelBuildParameters{
+        {"INPUT_TYPE", input_dtype == "half" ? "half" : "float"},
+        {"OUTPUT_TYPE", output_dtype == "half" ? "half" : "float"},
+    };
+
+    kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+    kernel.l_wk.push_back(xlocalsize);
+    kernel.l_wk.push_back(ylocalsize);
+
+    kernel.g_wk.push_back(xgridsize);
+    kernel.g_wk.push_back(ygridsize);
+
+    result.construction_params.push_back(kernel);
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& invoke_params) {
+            decltype(auto) kernel = handle_.Run(kernels.front());
+            decltype(auto) params =
+                invoke_params.CastTo<miopen::pad_constant_fwd_contiguous::InvokeParams>();
+
+            auto xdims    = params.xDesc->GetLengths();
+            auto ydims    = params.yDesc->GetLengths();
+            auto xstrides = params.xDesc->GetStrides();
+            auto ystrides = params.yDesc->GetStrides();
+
+            tensor_view_5d_t input_tv, output_tv;
+
+            for(size_t i = 0; i < 5; i++)
+            {
+                input_tv.size[i]   = xdims[i];
+                input_tv.stride[i] = xstrides[i];
+
+                output_tv.size[i]   = ydims[i];
+                output_tv.stride[i] = ystrides[i];
+            }
+
+            // Calculate output size (again)
+            size_t output_size = params.yDesc->GetElementSize();
+
+            kernel(params.x,
+                   params.y,
+                   input_tv,
+                   output_tv,
+                   params.padding,
```

Comment:
In my opinion, when the padding is passed to the GPU, it should be passed as a static array of length 10, like implemented in moDNN.

Reply:
I've created a padding type and changed the kernel input parameters